eng/common/pipelines/templates/steps/credscan.yml

```diff
@@ -0,0 +1,41 @@
+parameters:
+  SuppressionFilePath: 'eng/CredScanSuppression.json'
+  BaselineFilePath: ''
+  SourceDirectory: $(Build.SourcesDirectory)
+
+steps:
+- pwsh: |
+    New-Item -Path "${{ parameters.SourceDirectory }}" -Name "credscan.tsv"
+    if ("$(Build.Reason)" -eq 'PullRequest') {
+      New-Item -Path "${{ parameters.SourceDirectory }}" -Name "changedFiles" -ItemType "directory"
+      (git diff "origin/$(System.PullRequest.TargetBranch)" HEAD --name-only)
+      | ForEach-Object { Add-Content -Path "${{ parameters.SourceDirectory }}/credscan.tsv" -Value "${{ parameters.SourceDirectory }}/$_"}
+    }
+    else {
+      Set-Content "credscan.tsv" -Value "${{ parameters.SourceDirectory }}/sdk/${{ parameters.ServiceDirectory }}"
```

Comment:
Looks like you are expecting a ServiceDirectory parameter as well. We should either add the parameter to the list or see if we can just reuse SourceDirectory.

Reply:
Better to add service directory as they are different value